handlers/graph.go

```diff
@@ -937,8 +962,10 @@ func promQuery(query string, queryTime time.Time, api v1.API) model.Vector {
 	query = fmt.Sprintf("round(%s,0.001)", query)
 	log.Debugf("Graph query:\n%s@time=%v (now=%v, %v)\n", query, queryTime.Format(graph.TF), time.Now().Format(graph.TF), queryTime.Unix())
 
+	promtimer := internalmetrics.GetPrometheusProcessingTimePrometheusTimer("Graph-Generation")
```

Comment:
There's an inception flavour here :)

Reply:
I think you are referring to the duplication of the words "Prometheus" and "Time(r)" here.

They are, in fact, referring to two different things though, and should make sense when you understand the pattern of the naming of the functions in this utility file.

The suffix "PrometheusTimer" is the same suffix for all the utility functions that return a timer that is used for collecting Prometheus metrics.

The prefix "Get" is also used for all utility functions that "get" one of these timers.

The "PrometheusProcessingTime" is the actual metric - and it is referring to the time it takes for the Prometheus server to process a query request.

So, yes, it seems like it is duplicating the words in the name, but they truly are referring to different things (the Prometheus "server" and the Prometheus "client") and they follow the pattern of the rest of the getter utility methods.

Reply:
No problem with the naming, I understand the duplication, I was just referring to using prometheus (server) to monitor prometheus (client)... that's fine! :)